apps/provider-console/src/components/deployments/DeploymentDetails.tsx

```diff
@@ -43,39 +44,35 @@ interface DeploymentDetails {
 }
 
 export const DeploymentDetails: React.FC<DeploymentDetailsProps> = ({ dseq, owner }) => {
-  const [deploymentDetails, setDeploymentDetails] = useState<DeploymentDetails | null>(null);
-  const [loading, setLoading] = useState<boolean>(true);
-  const [error, setError] = useState<string | null>(null);
+  const {
+    data: deploymentDetails,
+    isLoading,
+    error
+  } = useQuery({
+    queryKey: ["deployment", owner, dseq],
+    queryFn: async () => {
+      const [response, latestBlocks]: any = await Promise.all([consoleClient.get<any>(`v1/deployment/${owner}/${dseq}`), consoleClient.get<any>(`/v1/blocks`)]);
```

Comment:
Feel free to refactor this or not, but just for you info, we usually extract this logic is separate files to have the components only have render logic and fetching in other like this: https://github.com/akash-network/console/tree/main/apps/deploy-web/src/queries

I noticed other components who had the same fetching code in a useEffect but didn't add a comment because I thought it was implicit to the same. Anyways, it's up to you if you want to refactor this now or later, but it would be great if we had the same structure.

Reply:
It's actually a great idea. I will refactor this today on the same branch; I don't want to delay this.

Reply:
I refactored it and moved all the fetch queries except the wallet provider context. I am figuring out how to approach this, but other than that, everything is moved.

Reply:
Sounds good!